pkg/golinters/gofumpt.go

```diff
@@ -0,0 +1,76 @@
+package golinters
+
+import (
+	"bytes"
+	"fmt"
+	"go/token"
+	"io/ioutil"
+	"sync"
+
+	"golang.org/x/tools/go/analysis"
+	"mvdan.cc/gofumpt/format"
+
+	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
+	"github.com/golangci/golangci-lint/pkg/lint/linter"
+	"github.com/golangci/golangci-lint/pkg/result"
+)
+
+const gofumptName = "gofumpt"
+
+func NewGofumpt() *goanalysis.Linter {
+	var mu sync.Mutex
+	var resIssues []goanalysis.Issue
+
+	analyzer := &analysis.Analyzer{
+		Name: gofumptName,
+		Doc:  goanalysis.TheOnlyanalyzerDoc,
+	}
+	return goanalysis.NewLinter(
+		gofumptName,
+		"Gofumpt checks whether code was gofumpt-ed.",
+		[]*analysis.Analyzer{analyzer},
+		nil,
+	).WithContextSetter(func(lintCtx *linter.Context) {
+		analyzer.Run = func(pass *analysis.Pass) (interface{}, error) {
+			var fileNames []string
+			for _, f := range pass.Files {
+				pos := pass.Fset.PositionFor(f.Pos(), false)
+				fileNames = append(fileNames, pos.Filename)
+			}
+
+			var issues []goanalysis.Issue
+
+			for _, f := range fileNames {
+				input, err := ioutil.ReadFile(f)
+				if err != nil {
+					return nil, fmt.Errorf("unable to open file %s: %w", f, err)
+				}
+				output, err := format.Source(input, "")
+				if err != nil {
+					return nil, fmt.Errorf("error while running gofumpt: %w", err)
+				}
+				if !bytes.Equal(input, output) {
+					issues = append(issues, goanalysis.NewIssue(&result.Issue{
+						FromLinter: gofumptName,
+						Text:       "File is not `gofumpt`-ed",
+						Pos: token.Position{
```

Comment:
There is a `Replacement` parameter to automatically fix issues, which is very useful for gofumpt.

Reply:
If I'm not mistaken, gofumpt provides only as a public function a way to get the gofumpt-ed version of the file (as a `[]byte`).
With `Replacement` you mean I should include the gofumpt-ed version of the whole file in the issue?

Reply:
@tetafro Am I correct?

Reply:
> gofumpt provides only as a public function a way to get the gofumpt-ed version of the file

Yeah, sorry, I didn't think of it.

> With Replacement you mean I should include the gofumpt-ed version of the whole file in the issue?

I'm not sure if it's a right thing to do. Someone more experienced with golangci-lint architecture should answer.

---

Anyway, if there's no easy way to do this, just skip my first comment.